src/main/java/me/playbosswar/com/utils/gson/GsonDate.java

```diff
@@ -19,12 +22,14 @@ public Date deserialize(JsonElement jsonElement, Type type,
             return formatter.parse(jsonElement.getAsString());
         } catch(ParseException ex1) {
             // Check if maybe it's in the old format
+            Bukkit.getLogger().log(Level.WARNING, "Could not parse date in new format, trying old format");
             try {
                 SimpleDateFormat oldFormatter = new SimpleDateFormat("MMM d, yyyy, H:mm:ss a");
                 return oldFormatter.parse(jsonElement.getAsString());
             } catch(ParseException ex2) {
+                Bukkit.getLogger().log(Level.WARNING, "Could not parse date in new format, trying old format");
                 try {
-                    SimpleDateFormat oldFormatter = new SimpleDateFormat("MMM d, yyyy, HH:mm:ss a");
+                    SimpleDateFormat oldFormatter = new SimpleDateFormat("MMM d, yyyy, hh:mm:ss a", Locale.ENGLISH);
```

Comment:
I would prefer to have another try/catch clause here instead of changing this one. Certain users do have this HH:mm:ss format. It starts to get a bit clumsy with all these try/catches so if you have a cleaner way of checking this go ahead

Reply:
i would like to use a List<string> to collect those formats.
```java
    @Override
    public Date deserialize(JsonElement jsonElement, Type type,
                            JsonDeserializationContext jsonDeserializationContext) throws JsonParseException {
        SimpleDateFormat formatter = new SimpleDateFormat(FORMAT);
        List<String> formats = List.of(
                "MMM d, yyyy, H:mm:ss a", 
                "MMM d, yyyy, HH:mm:ss a", 
                "MMM d, yyyy, HH:mm:ss", 
                "MMM d, yyyy, hh:mm:ss a"); 
        for (String dataFormat : formats) {
            try {
                SimpleDateFormat oldFormatter = new SimpleDateFormat(dataFormat, Locale.ENGLISH);
                return oldFormatter.parse(jsonElement.getAsString());
            }
            catch (ParseException ex) {
                // ignore
            }
        }
        throw new JsonParseException("Can not parse data: " + jsonElement.getAsString());
    }
```

Reply:
that's the final version:
    @Override
    public Date deserialize(JsonElement jsonElement, Type type,
                            JsonDeserializationContext jsonDeserializationContext) throws JsonParseException {
        List<SimpleDateFormat> formats = List.of(
                new SimpleDateFormat(FORMAT),
                new SimpleDateFormat("MMM d, yyyy, H:mm:ss a"),
                new SimpleDateFormat("MMM d, yyyy, HH:mm:ss a"),
                new SimpleDateFormat("MMM d, yyyy, HH:mm:ss"),
                new SimpleDateFormat("MMM d, yyyy, hh:mm:ss a"));
        for (SimpleDateFormat dataFormat : formats) {
            try {
                return dataFormat.parse(jsonElement.getAsString());
            } catch (ParseException ex) {
                // ignore
            }
        }
        throw new JsonParseException("Can not parse data: " + jsonElement.getAsString());
    }